docs/csharp/language-reference/builtin-types/nullable-reference-types.md

```diff
@@ -0,0 +1,74 @@
+---
+title: "Nullable reference types - C# reference"
+description: Learn about C# nullable reference types and how to use them
+ms.date: 04/06/2020
+---
+# Nullable reference types (C# reference)
+
+> [!NOTE]
+> This article covers nullable reference types. You can also declare [nullable value types](nullable-value-types.md).
+
+Nullable reference types are available beginning with C# 8.0, in code that has opted in to a *nullable aware context*. Nullable reference types, the null static analysis warnings, and the null forgiveness operator are optional language features. All are turned off by default. A *nullable context* is controlled at the project level using build settings, or in code using pragmas.
+
+ In a nullable aware context:
+
+- A reference type `T` must be initialized to non-null, and may never be assigned a value that may be `null`.
+- A reference type `T?` may be initialized or assigned to `null`, but is required to be checked against `null` before de-referencing.
+- A variable `m` of type `T?` is considered to be non-null if the *null forgiveness operator* is appended to the variable name, as in `m!`.
+
+The distinctions between a non-nullable reference type `T` and a nullable reference type `T?` are performed by the compiler's interpretation of the preceding rules. A variable of type `T` and a variable of type `T?` are represented by the same type in IL and at runtime. The following example declares a non-nullable string, a nullable string, and the uses the null forgiveness operator to assign a value to a non-nullable string:
+
+:::code language="csharp" source="snippets/NullableReferenceTypes.cs" id="SnippetCoreSyntax":::
+
+The variable `notNull` and `nullable` are both represented by the <xref:System.String> type. Because the non-nullable and nullable types are both stored as the same type, there are several locations where using a nullable reference type isn't allowed. In general, a nullable reference type can't be used as a base class or implemented interface. A nullable reference type can't be used in any object creation or type testing expression. A nullable reference type can't be the type of a member access expression. The following examples show these constructs:
```

Comment:
> Because the non-nullable and nullable types are both stored as the same type

That reads not clear. Can that whole sentence be removed, and then just list the locations where nullable reference types aren't allowed?

```suggestion
The variables `notNull` and `nullable` are both represented by the <xref:System.String> type. Because the non-nullable and nullable types are both stored as the same type, there are several locations where using a nullable reference type isn't allowed. In general, a nullable reference type can't be used as a base class or implemented interface. A nullable reference type can't be used in any object creation or type testing expression. A nullable reference type can't be the type of a member access expression. The following examples show these constructs:
```

Reply:
I'll take this suggestion, and consider making a list.